### DISCUSSION TITLE
Difference between ref={register} and ref={register()} ?

### DISCUSSION THREAD

**Discussion by jvgeee** [react-hook-form/react-hook-form#3396] [Q&A] [answered]
I have a wizard that saves a user's answers + displays different steps / questions based on their answers, and I've had to use `register()` to consistently be able to get my default values working.

Just wondering what the difference between `register` and `register()` is => is there a performance hit with the latter? Any reason to not use `register()` all the time?

**Comment by bluebill1049** [comment, marked as answer]
you can use `register() ` all the time, no difference :)

**Reply by jvgeee** [reply]
awesome! Thanks for the quick response mate, appreciate your awesome product and all the hard work you to do answer questions here.

**Reply by bluebill1049** [reply]
no worries, mate. I believe the more I help, the others will starts help each other in the long run.

**Reply by mordechaim** [reply]
For `useFieldArray()` it only works properly with `register()` though.

**Reply by bluebill1049** [reply]
yes, it's in the doc. we are fixing this inconsistency in the next major version.
